### PR TITLE
Refactor common.launch and main launch scripts to avoid duplicating them in ow_sim_tests

### PR DESCRIPTION
**Pull request by Samahu**
## Linked Issues:
| EPIC ⚡ | [OCEANWATER-684 / Automated build and tests](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-684) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-741 / Check if it possible to avoid copying empty_world.launch in ow_sim_tests](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-741) |
| Jira Ticket 🎟️ | [OCEANWATER-742 / Refactor common.launch and other launch scripts to avoid duplicating them in ow_sim_tests](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-742) |
| Github :octocat:  | #149  |

## Summary of Changes
* Refactor `common.launch` to enable the following:
   - specify an alternative `empty_world.launch` launch file for spawning gazebo
   - allow suppressing rqt_gui
   - allow suppressing rviz
* Propagate these flags to main launch files (atacama_y1a.launch, europa_workspace.launch, ..)
* Feed in the **fps_mointor gui plugin** through a custom `empty_world.launch`
   - This now only injected when launching the **fps_monitor** test
* Made the test_minimum_fps test more robust by adding one more check
* Revised the minimum frame rate for the fps_monitor from 30 to 50

## Test
* Same procedures (exclude `arm_check.test`) described in #149 with the exception that RVIZ and RQT GUI should not show up when launching test
* You may also launch the main simulation launch file and verify that both RVIZ and RQT GUI launch _by default_.
```bash
roslaunch ow atacama_y1a.launch
```

### CODE REVIEW

**Comment by Samahu** [comment]
Added @AstroStucky to the review since he touched on in the issue of having too many duplicate files as part of #149

**Comment by AstroStucky** [comment]
The tests worked great. rostest did not launch rviz or rqt, but the atacama_y1a.launch file did.

The names of the arguments seem inconsistent with each other since they both provide they same function of either showing or not showing a GUI. I think their usage would be easier to remember if they were instead `rqt_gui` and `rviz_gui`, or something similar.

**Comment by AstroStucky** [comment]
That said, I'm fine accepting it without the suggested change, especially since I don't fully know what else not having rviz or rqt gui running affects in the simulation.

Great work!

**Review by AstroStucky — APPROVED** [review]
That said, I'm fine approving this without that suggested change; especially since I don't fully know what else not having rviz or rqt gui running affects in the simulation.

Great work!

**Comment by Samahu** [comment]
> The tests worked great. rostest did not launch rviz or rqt, but the atacama_y1a.launch file did.
> 
> The names of the arguments seem inconsistent with each other since they both provide they same function of either showing or not showing a GUI. I think their usage would be easier to remember if they were instead `rqt_gui` and `rviz_gui`, or something similar.

I had same idea but we don't have a convention around this. **gzclient** is another argument of same purpose.

File: ow/launch/common.launch [hunk 1]
```diff
@@ -77,19 +81,13 @@
...
-  </group-->
-
   <!-- == Start rqt with a short delay so most topics will be visible and we won't need to refresh widgets ====== -->
   <!-- Starting without a .perspective file will reload the last used configuration. -->
   <!--node pkg="rqt_gui" type="rqt_gui" name="rqt_gui"
     launch-prefix="bash -c 'sleep 5; $0 $@' "/-->
   <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui"
     launch-prefix="bash -c 'sleep 5; $0 $@' "
-    args="&#45;&#45;perspective-file $(find ow)/config/default.perspective">
+    args="&#45;&#45;perspective-file $(find ow)/config/default.perspective" if="$(arg rqt_gui)">
```
Comment (mogumbo) [inline ow/launch/common.launch:90]: #nit I like having the conditional on the first line after "<node" since everything that follows is dependent upon it.

**Review by mogumbo — APPROVED** [review]

File: ow/launch/common.launch (hunk 1 above)
Comment (Samahu) [reply ow/launch/common.launch]: addressed

**Comment by kmdalal** [comment]
I ran the FPS test twice and it failed both times.  This may be unrelated to this PR:
```
[ow_sim_tests.rosunit-minimum_fps/test_minimum_fps][FAILURE]--------------------
observed fps fell below the lowest threshold
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/usr/local/home/kdalal/ow/src/ow_simulator/ow_sim_tests/test/test_minimum_fps.py", line 59, in test_minimum_fps
    "observed fps fell below the lowest threshold")
  File "/usr/lib/python2.7/unittest/case.py", line 943, in assertGreater
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
--------------------------------------------------------------------------------


SUMMARY
 * RESULT: FAIL
 * TESTS: 1
 * ERRORS: 0
 * FAILURES: 1

```

File: ow_sim_tests/test/fps_monitor.test [hunk 2]
```diff
@@ -3,10 +3,14 @@
...
-    <include file="$(find ow_sim_tests)/launch/atacama_y1a.launch" />
+    <include file="$(find ow)/launch/atacama_y1a.launch">
+        <arg name="gazebo_launch_file" default="$(find ow_sim_tests)/launch/empty_world_inject_fps_monitor_gui_plugin.launch"/>
+        <arg name="rqt_gui" value="false"/>
+        <arg name="use_rviz" value="false"/>
+    </include>
 
     <test test-name="minimum_fps" pkg="ow_sim_tests" type="test_minimum_fps.py">
-        <param name="min_fps" value="30.0"/>
+        <param name="min_fps" value="50.0"/>
```
Comment (kmdalal) [inline ow_sim_tests/test/fps_monitor.test:13]: Actually, this increase may have caused my test failures.  What's the motivation?

Comment (Samahu) [reply ow_sim_tests/test/fps_monitor.test:13]: I always get 60 on my machine, so I thought I can make the test more strict. Do you know the average FPS that you get on your machine (when you have full view of atacama and the lander is standstill)?

Comment (Samahu) [reply ow_sim_tests/test/fps_monitor.test]: Ok, I have adjusted the **min_fps** to 45 based on the feedback from Terry that his old machine currently reports 47 fps.
I have also updated the failure message to report the observed frame rate.

**Comment by kmdalal** [comment]
My frame rate has been around 12, and I discovered that the reason was an old 'gzclient' process that must have been left over from my scaling fix experiments.  After killing that, and testing many times, I typically get frame rates between 45-50.  However, once I got 15 for some reason and the test failed.   This is a highly configured native Ubuntu laptop, and the load is not high.

**Comment by Samahu** [comment]
> My frame rate has been around 12, and I discovered that the reason was an old 'gzclient' process that must have been left over from my scaling fix experiments. After killing that, and testing many times, I typically get frame rates between 45-50. However, once I got 15 for some reason and the test failed. This is a highly configured native Ubuntu laptop, and the load is not high.

Do you always have this happen? Or it was just a one time thing.

**Comment by Samahu** [comment]
If it is a one time thing, then I would say this is probably due to a external event on your machine. For example another process needed access to the GPU at some point (chrome and other browsers make use of the GPU).

**Comment by kmdalal** [comment]
Did several more tests, with frame rates 45-50; once it was 60.  I'll consider the odd 15 a one-off occurrence, per your explanation - thanks.

**Review by kmdalal — APPROVED** [review]
My occasional slow FPS rates are a problem unrelated to this work.  This is a good refactoring!

**Pull request merged**